checks/network_checks.py

```diff
@@ -18,6 +18,8 @@
 MAX_LOOP_ITERATIONS = 1000
 FAILURE = "FAILURE"
 
+_global_current_pool_size = 0
```

Comment:
Could this just be a property on the `NetworkCheck` class? Should save the need for the `globals` and tie it to a logically connected class.